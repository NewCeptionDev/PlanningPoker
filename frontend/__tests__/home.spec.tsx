import { act, render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import Home from 'src/app/page'

describe('Home', () => {
  it('should render homepage unchanged', () => {
    const { container } = render(<Home />)
    expect(container).toMatchSnapshot()
  })

  it('should render all important components', () => {
    render(<Home />)
    expect(screen.getByTestId('logo')).toBeInTheDocument()
    expect(screen.getByTestId('headline')).toBeInTheDocument()
    expect(screen.getByTestId('theme-switcher')).toBeInTheDocument()
    expect(screen.getByTestId('lobby-join')).toBeInTheDocument()
    expect(screen.getByTestId('lobby-create')).toBeInTheDocument()
    expect(screen.getByTestId('footer')).toBeInTheDocument()
  })

  it('should show custom card input when custom card group is selected', () => {
    render(<Home />)

    expect(screen.queryByTestId('lobby-create-custom-cards')).not.toBeInTheDocument()

    act(() => {
      const select = screen.getByTestId('cardGroupSelect')
      // @ts-expect-error value is defined
      select.value = 'Custom'
      select.dispatchEvent(new Event('change', { bubbles: true }))
    })

    expect(screen.getByTestId('lobby-create-custom-cards')).toBeInTheDocument()
  })
})
