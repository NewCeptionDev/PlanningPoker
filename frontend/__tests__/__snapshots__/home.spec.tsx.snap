// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render homepage unchanged 1`] = `
<div>
  <main
    class="h-screen light-bg dark:dark-bg"
  >
    <div
      class="flex flex-row justify-evenly items-center h-[30vh]"
    >
      <div
        class="w-1/5 m-8"
      />
      <div
        class="w-3/5 flex flex-col items-center justify-center"
      >
        <img
          alt="logo"
          data-nimg="1"
          data-testid="logo"
          decoding="async"
          height="100"
          loading="lazy"
          src="/_next/image?url=%2Flogo_large.png&w=256&q=75"
          srcset="/_next/image?url=%2Flogo_large.png&w=128&q=75 1x, /_next/image?url=%2Flogo_large.png&w=256&q=75 2x"
          style="color: transparent;"
          width="100"
        />
        <h1
          class="text-6xl text-center"
          data-testid="headline"
        >
          Planning
          <br />
          Poker
        </h1>
      </div>
      <div
        class="w-1/5 flex justify-end m-8 self-start"
      >
        <div
          class="grid place-items-center h-full relative"
          data-testid="theme-switcher"
        >
          <input
            class="hidden toggle--checkbox"
            data-testid="theme-switcher-toggle"
            id="toggle"
            type="checkbox"
          />
          <label
            class="toggle--label"
            data-testid="theme-switcher-label"
            for="toggle"
          >
            <span
              class="toggle--label-background"
              data-testid="theme-switcher-label-background"
            />
          </label>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col justify-center items-center h-[66vh]"
      id="content"
    >
      <div
        class="flex flex-col items-center"
        data-testid="lobby-join"
        id="lobby-join"
      >
        <h1
          class="mb-4"
        >
          Enter a Lobby Code
        </h1>
        <form
          class="flex flex-col items-center"
        >
          <input
            class="input"
            name="lobbyId"
            placeholder="Lobby ID"
            type="text"
          />
          <button
            class="btn"
            id="lobby-join-button"
            type="submit"
          >
            Join
          </button>
        </form>
      </div>
      <div
        class="w-1/3 h-0.5 self-center m-8 secondary"
        id="spacer"
      />
      <div
        class="flex flex-col items-center justify-center w-1/4"
        data-testid="lobby-create"
        id="lobby-create"
      >
        <h1
          class="mb-4"
        >
          Create a new Lobby
        </h1>
        <form
          class="flex flex-col items-center"
        >
          <input
            class="input"
            name="lobbyName"
            placeholder="Lobby Name"
            type="text"
          />
          <p
            class="mb-2"
          >
            Select the Cards to be used
          </p>
          <select
            class="input flex"
            data-testid="cardGroupSelect"
            id="cardGroupSelect"
            name="cardGroup"
          >
            <option
              class="w-[90%] max-w-[90%] overflow-ellipsis"
              value="Simple"
            >
              Simple
               (1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
            </option>
            <option
              class="w-[90%] max-w-[90%] overflow-ellipsis"
              value="Fibonacci"
            >
              Fibonacci
               (1, 2, 3, 5, 8, 13, 21, 34, 55)
            </option>
            <option
              class="w-[90%] max-w-[90%] overflow-ellipsis"
              value="Modified Fibonacci"
            >
              Modified Fibonacci
               (1, 2, 3, 5, 8, 13, 20, 40, 100)
            </option>
            <option
              class="w-[90%] max-w-[90%] overflow-ellipsis"
              value="Custom"
            >
              Custom
            </option>
          </select>
          <button
            class="btn"
            id="lobby-create-button"
            type="submit"
          >
            Create
          </button>
        </form>
      </div>
    </div>
    <div
      class="flex flex-row justify-center h-[2vh]"
      data-testid="footer"
    >
      <p>
        Made by 
        <a
          href="https://newception.dev"
        >
          @NewCeption
        </a>
      </p>
    </div>
  </main>
</div>
`;
